packages/string/src/utils/ansi-state-tracker.ts

```
@@ -0,0 +1,89 @@
+    // Track other formatting (bold, italic, etc.)
+    private activeFormatting: string[] = [];
+
+    /**
+     * Processes an escape sequence and updates the internal state
+     * @param sequence - The escape sequence to process
+     */
+    // eslint-disable-next-line sonarjs/cognitive-complexity
+    public processEscape(sequence: string): void {
+        // Extract the numeric code from the sequence
+        // eslint-disable-next-line no-control-regex,regexp/no-control-character
+        const match = /\u001B\[(\d+)m/.exec(sequence);
```

Biome (1.9.4)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)